source/grid/grid_tools.cc

```diff
@@ -5883,67 +5888,92 @@ namespace GridTools
       const std::vector<T> &                                 entities,
       const double                                           tolerance)
     {
+      std::vector<std::pair<unsigned int, unsigned int>> ranks_and_indices;
+      ranks_and_indices.reserve(entities.size());
+
+#ifdef DEAL_II_WITH_ARBORX
+      if (global_bboxes.size() == 1)
+        {
```

Comment:
@fdrmrc I think I was not clear what I wanted. I would prefer
```
#if defined(DEAL_II_WITH_ARBORX)
if(global_bboxes.size() == 1)
{
  // new code
}
else
{
  // old code
}
```

Reply:
My bad! I'm sorry but I don't fully understand what you mean with your snippet. Do you mean that if ArborX is available it would be used only when `global_bboxes.size()==1`?

Reply:
> Do you mean that if ArborX is available it would be used only when global_bboxes.size()==1

Yes.

Reply:
Thanks for the confirmation! PR updated.